app/controllers/api/addresses_controller.rb

```diff
@@ -7,8 +7,7 @@ class Api::AddressesController < Api::BaseController
   filter_resources_by :account_id
 
   def resource
-    @address = account.address if !super && account
-    @address
+    @address ||= account.try(:address) || super
```

Comment:
I think it is more correct now. It previously checked `super` in the trailing if, but did not use it.

Now it correctly tries to get the address for an account_id if specified, and then falls back on the address id.  I suppose it could check in the opposite order, but as the routes and representers are set up to use addresses only as belonging to users and accounts which can only have one address, the account nested route is the default.